core/server/models/user.js

```diff
@@ -324,6 +326,22 @@ User = ghostBookshelf.Model.extend({
         options = options || {};
         options.withRelated = _.union(options.withRelated, options.include);
 
+        if (data.email && data.id) {
+            return this.getByEmail(data.email, {id: data.id}).then(function then(user) {
+                if (user) {
+                    return Promise.reject(new errors.ValidationError(i18n.t('errors.models.user.userUpdateError.emailIsAlreadyInUse')));
+                }
+                return self.update.call(self, data, options);
+            });
+        } else {
+            return self.update.call(self, data, options);
+        }
+    },
+
+    update: function update(data, options) {
```

Comment:
with this change, we export a new function called `update`. So somebody could use in `models.User.update` instead of `models.User.edit`. I suggest you solve the calling problem with `Promise`. Something like (pseudo code!):

```
// CASE: email gets edited
if (data.email) {
  ops.push(this.getByEmail);
}

ops.push(function() {
   return ghostBookshelf.Model.edit.call....
})

return pipeline(ops);
```